main.go

```diff
@@ -396,6 +486,177 @@ func runInPod(name string, cmdToRun string, env []string, timeout int) ([]string
 	return lines[:len(lines)-1], timeout_reached
 }
 
+func selectNodes(step *Step, config Config, subsetPartition map[int][]int) []int{
+	/* Handle OnNode EndNode syntax */
+	if step.OnNode > 0 {
+		if step.Selection != nil {
+			fatal("Two node selection methods on test step")
+		}
+		if step.EndNode == 0 {
+			step.EndNode = step.OnNode
+			selected := make([]int, 1)
+			selected[0] = step.OnNode
+			return selected
+		}
+		return makeRange(step.OnNode, step.EndNode)
+	} else if step.Selection == nil {
+		fatal("No selection method on test step")
+	} else { /* Selection chooses nodes */
+		if (step.Selection.Range == nil && step.Selection.Percent == nil && step.Selection.Subset == nil) {
+			fatal("No selection method on test step")
+		} else if (step.Selection.Range != nil) {
+			return selectNodesRange(step, config)
+		} else if (step.Selection.Percent != nil) {
+			return selectNodesPercent(step, config)
+		} else /* Subset selection */ {
+			return selectNodesSubset(step, subsetPartition)
+		}
+	}
+	/* Should not get here, keeping compiler happy for now*/
+	fatal("Should not reach!")
+	return nil
+}
+
+func selectNodesRange(step *Step, config Config) []int{
+	var selection []int
+	if step.Selection.Range.Order == SEQUENTIAL {
+	  if step.Selection.Range.Start <= 0 || 
```

Comment:
@hsanjuan Noted, I will refactor into pre-validation and bubble up errors in the next changes.